crates/ty_python_semantic/resources/mdtest/type_properties/is_disjoint_from.md

```diff
@@ -87,6 +87,25 @@ static_assert(is_disjoint_from(memoryview, Foo))
 static_assert(is_disjoint_from(type[memoryview], type[Foo]))
 ```
 
+## Specialized `@final` types
+
+```toml
+[environment]
+python-version = "3.12"
+```
+
+```py
+from typing import final
+from ty_extensions import static_assert, is_disjoint_from
+
+@final
+class Foo[T]:
+    def get(self) -> T:
+        raise NotImplementedError
+
+static_assert(not is_disjoint_from(Foo[int], Foo[str]))
```

Comment:
`int` and `str` are disjoint bases, so these two actually _should_ be disjoint (we're missing a check somewhere else). So this would be a better test here:

```suggestion
class A: ...
class B: ...

static_assert(not is_disjoint_from(Foo[A], Foo[B]))
```